src/renderer/utils/FileMonitor.tsx

```diff
@@ -0,0 +1,264 @@
+import chokidar from 'chokidar';
+import { Subject } from 'rxjs';
+
+import { Event, Monitor } from './Monitor';
+
+// type Event = 'unlink' | 'unlinkDir' | 'add' | 'addDir' | 'change' | 'error';
+
+class FileMonitor implements Monitor {
+  //The main functionality of the file monitor is to report selected type of events using the Observable called subject
+  private folderPath: string;
+  //ignoredPath: string;
+  private events: Array<Event>;
+  private eventsWatching: Array<Event>;
+  private subject: Subject<[Event, string]>;
+  private addWatcher: chokidar.FSWatcher;
```

Comment:
You can just use a map to keep track of the handlers, no? I don't think there needs to be that many fields.